src/app/overview/widgets/coldstake/coldstake.component.ts

```diff
@@ -1,27 +1,64 @@
 import { Component, OnInit } from '@angular/core';
+import { Log } from 'ng2-logger';
 
 import { RPCService } from '../../../core/rpc/rpc.service';
 import { ModalsService } from '../../../modals/modals.service';
 
+import { Amount } from '../../../shared/util/utils';
+
 @Component({
   selector: 'app-coldstake',
   templateUrl: './coldstake.component.html',
   styleUrls: ['./coldstake.component.scss']
 })
 export class ColdstakeComponent implements OnInit {
 
+  /*  General   */
+  private log: any = Log.create('send.component');
   private coldStakingEnabled: boolean = undefined;
 
+  private progress: Amount = new Amount(0, 2);
+  get coldstakeProgress(): number { return this.progress.getAmount() }
+
   constructor(
     private _modals: ModalsService,
     private _rpc: RPCService
   ) {
     this._rpc.state.observe('ui:coldstaking').subscribe(status => this.coldStakingEnabled = status);
+
+    this.rpc_progressLoop();
   }
 
   ngOnInit() {
   }
 
+  /** calls listunspent, then calculate progress. */
+  private rpc_progressLoop() {
+    this._rpc.call('listunspent')
+    .subscribe(
+      (response: Array<any>) => {
+        let activeCount = 0;
+        let totalCount = 0;
+
+        response.forEach((output) => {
+          console.log(output.amount);
+          totalCount += output.amount;
+
+          if (output.coldstaking_address !== undefined) {
+            activeCount += output.amount;
+          }
+          this.log.d(`activeCount=${activeCount} totalCount=${totalCount}`);
+          this.progress = new Amount((activeCount / totalCount) * 100, 2);
+        });
+      },
+      // TODO: Handle error appropriately
+      error => this.log.er('rpc_progressLoop: listunspent failed', error));
+
+    if (this.coldstakeProgress < 100) {
+      setTimeout(this.rpc_progressLoop.bind(this), 1000);
+    }
+  }
+
   isColdStakingEnabled() {
     return this.coldStakingEnabled;
```

Comment:
this is also unnecessary.. uses more memory for no reason, as it will copy the variable into a new memory space before returning the result.. Lets make the property public rather